### PR TITLE
Guided Tours: a few tweaks to the documentation

### PR DESCRIPTION
**Pull request by lsinger**
As the Guided Tours framework has evolved, capabilities have been added and new insights gained. Those have not always documented, though. This PR tweaks the Guided Tours documentation a bit. 

To test: read the changes and evaluate whether they make enough sense.

### CODE REVIEW

File: client/layout/guided-tours/docs/DEBUGGING.md [hunk 1]
```diff
@@ -11,7 +11,13 @@ In the following example, the tours `gdocsIntegrationTour` and `editorInsertMenu
 
 ![Tour seen](./img/tour-seen.png)
 
-Currently, there is no systematic way of clearing a user's history for debugging. One way you can clear it is by applying the [`clean-history.patch`](./patch/clean-history.patch) to your working tree, load the page to clean the history, and remove it again.
+To clear a user's tour history, simply visit Calypso with `tour=reset` as a query argument. This will then call `resetGuidedToursHistory`:
```
Comment (nosolosw) [inline client/layout/guided-tours/docs/DEBUGGING.md]: Doesn't it also require to clear the application browser state? I couldn't trigger a tour I've had seen by using `tour=reset` until I delete my local application state in chrome.

Comment (lsinger) [reply client/layout/guided-tours/docs/DEBUGGING.md]: Actually it did work like that at some point, but it's not working anymore. Investigating.

File: client/layout/guided-tours/docs/API.md [hunk 2]
```diff
@@ -43,6 +45,7 @@ Step is a React component that defines a single Step of a tour. It is represente
...
 * `children`: (node) Contents of the step. Usually a paragraph of instructions and some controls for the tour. See below for available options. Note that all text content needs to be wrapped in `<p>` so it gets proper styling.
+* `onTargetDisappear`: (function, optional) In some cases the target that a step points to will disappear after Guided Tours has rendered a step. In those cases we can end up with steps unhelpfully pointing at (0, 0). The `onTargetDisappear` prop takes a function that will be called when the framework can't find the target anymore. The function will be passed an object containing two functions: `quit` and `next`. In your callback you can decide how to handle the disappearing target and then use those functions to either quit the whole tour or proceed to the next step. This is e.g. used in `checklist-site-icon-tour.js`.
```
Comment (taggon) [inline client/layout/guided-tours/docs/API.md]: Much better explanation than mine. 👍

File: client/layout/guided-tours/docs/DEBUGGING.md (hunk 1 above)
Comment (lsinger) [reply client/layout/guided-tours/docs/DEBUGGING.md]: Logged an issue here: #23598.

**Comment by lsinger** [comment]
d813971 removes the issue identified by @nosolosw.

**Pull request merged**